### PR TITLE
Update TRT-LLM backend build and install TRT-LLM within container

### PR DESCRIPTION
**Pull request by krishung5**

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -1113,6 +1113,9 @@ def create_dockerfile_linux(
       && rm -f ${TRT_ROOT}/targets/${ARCH}-linux-gnu/lib/libnvinfer*.a \\
           ${TRT_ROOT}/targets/${ARCH}-linux-gnu/lib/libnvonnxparser_*.a
 
+# Install TensorRT-LLM
+RUN python3 -m pip install /opt/tritonserver/backends/tensorrtllm/tensorrt_llm-*.whl -U --pre --extra-index-url https://pypi.nvidia.com
```
Comment (mc-nv) [inline build.py:1117]: I would suggest to remove this file, as it takes ~ 1.2 Gb
```suggestion
RUN python3 -m pip install /opt/tritonserver/backends/tensorrtllm/tensorrt_llm-*.whl -U --pre --extra-index-url https://pypi.nvidia.com \\
        && rm -fv /opt/tritonserver/backends/tensorrtllm/tensorrt_llm-*.whl

RUN find /usr -name libtensorrt_llm.so -exec dirname {} \; > /etc/ld.so.conf.d/tensorrt-llm.conf \\
        && ldconfig
```

File: build.py [hunk 2]
```diff
@@ -1708,6 +1711,34 @@ def tensorrtllm_prebuild(cmake_script):
...
+    # TODO: Update the CMakeLists.txt of TRT-LLM backend to install the artifacts to the correct location
+    cmake_destination_dir = os.path.join(repo_install_dir, "backends/tensorrtllm")
+    cmake_script.mkdir(cmake_destination_dir)
+    # Copy over the TRT-LLM wheel for later installation
+    cmake_script.cp(
+        os.path.join(tensorrtllm_be_dir, "tensorrt_llm", "build", "tensorrt_llm-*.whl"),
+        cmake_destination_dir,
+    )
+
+    # Copy over the TRT-LLM backend libraries
```
Comment (mc-nv) [inline build.py:1729]: If we produce only libraries, we don't need to specify each one, we can copy all `*so` files out of directory.

File: build.py [hunk 3]
```diff
@@ -1708,6 +1714,30 @@ def tensorrtllm_prebuild(cmake_script):
...
+    cmake_script.cp(
+        os.path.join(tensorrtllm_be_dir, "build", "libtriton_tensorrtllm*.so"),
+        cmake_destination_dir,
+    )
+    cmake_script.cp(
+        os.path.join(tensorrtllm_be_dir, "build", "triton_tensorrtllm_worker"),
```
Comment (mc-nv) [inline build.py:1738]: is this a correct value, i thought target file will be named as `triton_tensorrtllm_worker.so` ?

Comment (krishung5) [reply build.py:1738]: Yes, this should be correct. The `triton_tensorrtllm_worker` is not a .so:
```
root@sjc-dc2-dgx1-7:/opt/tritonserver/tensorrtllm_backend/ci/L0_backend_trtllm# ls -l /opt/tritonserver/backends/tensorrtllm/
total 604
-rwxrwxrwx 1 triton-server triton-server  96928 Apr 12 19:10 libtriton_tensorrtllm.so
-rwxrwxrwx 1 triton-server triton-server 490168 Apr 12 19:10 libtriton_tensorrtllm_common.so
-rwxrwxrwx 1 triton-server triton-server  28432 Apr 12 19:10 triton_tensorrtllm_worker
```

**Pull request closed without merging**